Changed QWidget -> QMainWindow to allow the application menu bar. QgsInterface returns a new QToolBar when addToolBar is called

JaumeFigueras opened the pull request:
Proposal to solve issue #13

JaumeFigueras commented:
I'm quite new to github. I wanted to create two PR, but my two commits appear in one PR. Also I detected that when a plugin calls the addToolBar function from iface, None was returned. But if this function is called a new toolbar has to be returned to the plugin abling it to add actions.

Joonalai reviewed (changes requested):
Thank you very much for your contribution, it is much appreciated!!

Don't worry about PRs, this is fine if you just modify the title of the PR to address both changes :)

I added some minor comments but other than that, can you just install pre-commit with:

```
pre-commit install
``` 
and that should fix all the code style issues automatically during the next commit. I should add pre-commit instructions to the README.md as well.

JaumeFigueras commented:
Hello @Joonalai,
I accepted your suggestions, added the `pre-commit install` and changed the PR title as requested.
Hope everything is OK now.
Thanks for your help!!

Joonalai commented on src/pytest_qgis/qgis_interface.py:
```diff
@@ -204,13 +213,15 @@ def removeToolBarIcon(self, action: QAction) -> None:
         """
-        pass
+        toolbar = QToolBar(name, parent=self._mainWindow)
+        self._toolbars.append(toolbar)
```
This will fail since self._toolbars is a dictionary. Use:
```suggestion
        self._toolbars[name] = toolbar
```

Joonalai reviewed (changes requested):
Great! Just one more tiny detail :)

JaumeFigueras commented:
Thank you for your hints.
